app/controllers/api/v2/ansible_playbooks_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    # API controller for Ansible Roles
+    class AnsiblePlaybooksController < ::Api::V2::BaseController
+      include ::Api::Version2
+
+      resource_description do
+        api_version 'v2'
+        api_base_url '/ansible/api'
+      end
+
+      before_action :find_proxy, :only => [:fetch, :sync]
+      before_action :create_importer, :only => [:fetch, :sync]
+
+      api :PUT, '/ansible_playbooks/sync', N_('Sync Ansible playbooks')
+      param :proxy_id, :identifier, :required => true, :desc => N_('Smart Proxy to sync from')
+      param :playbook_names, Array, N_('Ansible playbook names to be synced')
+      def sync
+        new_playbooks = @importer.import_playbook
+        @imported = new_playbooks
+      end
+
+      api :GET, '/ansible_playbooks/fetch', N_('Fetch Ansible playbooks available to be synced')
+      param :proxy_id, :identifier, N_('Smart Proxy to fetch from'),
+            :required => true
+      def fetch
+        fetched =  @importer.fetch_playbooks_names
+        respond_to do |format|
+          format.json do
```

Comment:
I do not think you need those 2 blocks as API controllers respond with json by default...